aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/migrations/ModelMigration.java

```diff
@@ -15,9 +15,12 @@
 
 package com.amplifyframework.datastore.storage.sqlite.migrations;
 
+import com.amplifyframework.annotations.InternalApiWarning;
+
 /**
  * Interface that defines the contract of an in-place model migration.
  */
+@InternalApiWarning
```

Comment:
I'd prefer that we don't go adding this annotation to existing public classes. It can give the impression later on that this annotation has always been there and that we are free to break the contract, which isn't the case.